packages/block-library/src/navigation-link/edit.js

```diff
@@ -232,16 +233,10 @@ export const updateNavigationLinkBlockAttributes = (
 		kind: newKind = originalKind,
 		type: newType = originalType,
 	} = updatedValue;
-	const normalizedTitle = title.replace( /http(s?):\/\//gi, '' );
+
 	const normalizedURL = url.replace( /http(s?):\/\//gi, '' );
-	const escapeTitle =
-		title !== '' &&
-		normalizedTitle !== normalizedURL &&
-		originalLabel !== title;
 
-	const label = escapeTitle
-		? escape( title )
-		: originalLabel || escape( normalizedURL );
+	const label = decodeEntities( title ) || originalLabel || normalizedURL;
```

Comment:
The alternative here is using `.raw` instead of `.rendered` on the return value from the API. But I'm still not clear on what the difference is between the two fields in this context.

Reply:
Here's where the fields are constructed by the REST API

https://github.com/WordPress/wordpress-develop/blob/19abcf2c4640ca5660089a6ada5eafbf8e4a5fa3/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1779-L1791

It's basically as follows:

```php
$data['title']['rendered'] = get_the_title( $post->ID );
$data['title']['raw'] = $post->post_title;
```

Essentially [rendered passes through default filters](https://github.com/WordPress/wordpress-develop/blob/613cac5a2d809a4078b23945a6c57750bdebd9b0/src/wp-includes/default-filters.php#L180-L182) whereas raw doesn't:

Reply:
I've stuck with `rendered` but used `decodeEntities` at point of output rather than when saving the attrs.